crates/oxc_transformer/src/es2018/object_rest_spread.rs

```diff
@@ -979,7 +979,9 @@ impl<'a> ObjectRestSpread<'a, '_> {
             }
             // `let { [`a`], ... rest }`
             PropertyKey::TemplateLiteral(lit) if lit.is_no_substitution_template() => {
-                let expr = Expression::TemplateLiteral(lit.clone_in(ctx.ast.allocator));
+                let quasis = ctx.ast.vec_from_iter(lit.quasis.iter().cloned());
+                let expressions = ctx.ast.vec_from_iter(lit.expressions.drain(..));
+                let expr = ctx.ast.expression_template_literal(lit.span, quasis, expressions);
```

Comment:
I think this one can be simplified.

`lit.is_no_substitution_template()` has returned `true` here.

https://github.com/oxc-project/oxc/blob/dbe61c571f0894471ce237b29c8fab77229be512/crates/oxc_ast/src/ast_impl/js.rs#L526-L528

```suggestion
                let quasis = ctx.ast.vec1(lit.quasis[0].clone());
                let expressions = ctx.ast.vec();
                let expr = ctx.ast.expression_template_literal(lit.span, quasis, expressions);
```

Reply:
Good catch! I didn't notice that check

Reply:
Note: Assuming that `is_no_substitution_template` is inlined here (which you'd expect, as it's so small), compiler should be able to optimize out the implicit bounds check in `lit.quasis[0]`, because it knows `self.quasis.len() == 1`.

That's an assumption, not a statement of fact - the compiler is sometimes much smarter than you'd expect, and sometimes much stupider! - but, generally speaking, you can expect the compiler to spot and make optimizations like that.